docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,256 @@
+- Defining SLO
+- Managing fine-tunes
+  - LoRA Adapters
+  - System Prompts
+  - Prompt Cache
+  - etc.
+- Managing rollout of adapters
+
+### Axioms 
+
+The API design is based on these axioms:
+
+- Pools of shared compute should be *discrete* for scheduling to properly work
+- Pod-level scheduling should not be handled by a high-level gateway 
+- Simple use cases should be simple to define (or are implicitly defined via reasonable defaults)
+- This solution should be composable with other Gateway solutions and flexible to fit customer needs
+- The MVP will heavily assume requests are done using the OpenAI spec, but open to extension in the future
+- The Gateway should route in a way that does not generate a queue of requests at the model server level
+
+The [PoC](https://youtu.be/NUBZg_uqqXk?si=v681EeYdGUGEVqQQ&t=1458) was focused on lower-level scheduling. And the API follows that similar logic, which lead to the proposal of the **BackendPool**.
+
+### BackendPool
+
+The BackendPool at its core is a logical grouping of compute, expressed in the form of Pods (typically model servers), akin to a K8s Service. The BackendPool would deploy its own routing, and offer administrative configuration to the Platform Admin. 
+
+ It is expected for the BackendPool to:
+ - Enforce fair consumption of resources across competing use cases
+ - Efficiently route requests across shared compute (as displayed by the PoC)
+ 
+It is _not_ expected for the BackendPool to:
+ - Enforce any common set of adapters or base models are available on the Pods
+ - Manage Deployments of Pods within the Pool
+ - Manage Pod lifecycle of pods within the pool 
+
+Additionally, any Pod that seeks to join a BackendPool would need to support a protocol, defined by LLM Instance Gateway, to ensure the Pool has adequate information to intelligently route requests.
+
+### ModelGroup
+
+A ModelGroup allows the UseCaseOwner to define:
+- Which LoRA adapter(s) to consume 
+  - ModelGroup allows for traffic splitting between adapters _in the same BackendPool_ to allow for new LoRA adapter versions to be easily rolled out 
+- SLO objectives for the UseCase
+- The Pools this UseCase is relevant to 
+
+### Spec
+
+**ModelGroup**
+```golang
+// ModelGroup represents a set of LLM use cases that are multiplexed onto one 
+// or more backend pools. This is generally owned by the "LLM Use Case Owner"
+// persona, which can be teams in an organization. Plural ModelUseCases are
+// allowed as a configuration convenience to the user. Allowing a user who
+// has multiple usecases across multiple pools (with the same config) to
+// specify the configuration exactly once, and deploy to many pools 
+// simultaneously, allowing for a simple config and single source of truth
+// for a given user. ModelUseCase names are unique for a given BackendPool,
+// if the name is reused, an error will be  shown on the status of a 
```

Comment:
Explicitly describe how the "reuser" is identified (i.e. the algorithm used to select which model group "wins").

Reply:
Took a stab PTAL